save-plugins/warn-plugin/src/commonTest/kotlin/org/cqfn/save/plugin/warn/WarnPluginTest.kt

```diff
@@ -240,11 +241,77 @@ class WarnPluginTest {
         fs.delete(tmpDir / "resource")
     }
 
+    @Test
+    @Suppress("TOO_LONG_FUNCTION")
+    fun `warn-plugin test for batchSize`() {
+        fs.write(fs.createFile(tmpDir / "resource")) {
+            write(
+                """
+                |Test1Test.java:4:6: Class name should be in PascalCase
+                |Test2Test.java:2:3: Class name should be in PascalCase
+                """.trimMargin().encodeToByteArray()
+            )
+        }
+        val catCmd = if (isCurrentOsWindows()) "type" else "cat"
+        performTest(
+            """
+                package org.cqfn.save.example
+                
+                // ;warn:4:6: Class name should be in PascalCase
+                class example {
+                    int foo = 42;
+                }
+            """.trimIndent(),
+            """
+                package org.cqfn.save.example
+                
+                // ;warn:2:3: Class name should be in PascalCase
+                class example2 {
+                    int foo = 42;
+                }
+            """.trimIndent(),
+            WarnPluginConfig(
+                "$catCmd ${tmpDir / "resource"}",
+                Regex("// ;warn:(\\d+):(\\d+): (.*)"),
+                Regex("(.+):(\\d+):(\\d+): (.+)"),
+                true, true, 2, 1, 2, 3, 1, 2, 3, 4
+            ),
+            GeneralConfig("", "", "", "")
+        ) { results ->
+            assertEquals(1, results.size)
+            assertTrue(results.single().status is Pass)
+        }
+        fs.delete(tmpDir / "resource")
+    }
+
     @AfterTest
     fun tearDown() {
         fs.deleteRecursively(tmpDir)
     }
 
+    private fun performTest(
```

Comment:
Instead of copy-pasting, you could refactor `performTest` to accept a collection of texts of arbitrary size, old implementation could call this one with a collection of single element.